packages/babel-plugin-transform-function-sent/test/fixtures/generator-kinds/class-method/expected.js

```diff
@@ -0,0 +1,12 @@
+function _skipFirstGeneratorNext(fn) { return function () { var it = fn.apply(this, arguments); it.next(); return it; }; }
+
+class Foo {
+  gen() {
+    return _skipFirstGeneratorNext(function* () {
```

Comment:
Copy-pasted the wrong thing 😅



Sorry, can you give me an example? The only way I can think of which works with `super.method()`/`new.target` is this:

```js
class A {
  *gen() {
    super.method(function.sent);
  }
}

// Becomes

let _gen;
let superCall_method;
class A {
  gen() {
    if (!_gen) _gen = _skipFirstGeneratorNext(function* () {
      let _functionSent = yield;
      super_callMethod(_functionSent);
    });

    super_callMethod = (...args) => super.method(...args);
    return _gen.apply(this, arguments);
  }
}
```

But it won't work if A#gen calls A#gen of another instance:
```js
class Base {
  method(x) {
    console.log(x + this.val);
  }
}

class A extends Base {
  constructor(val) {
    super();
    this.val = val;
  }

  *gen(x) {
    x && x.gen().next(7);
    super.method(function.sent);
  }
}

let n = new A(10);
let m = new A(100);
n.gen(m).next(1); // 107 11

// Becomes

class Base {
  method(x) {
    console.log(x + this.val);
  }
}

let _gen, super_callMethod;
class A extends Base {
  constructor(val) {
    super();
    this.val = val;
  }

  gen(x) {
    if (!_gen) _gen = _skipFirstGeneratorNext(function* (x) {
      let _functionSent = yield;
      x && x.gen().next(7);
       super_callMethod(_functionSent);
    });

    super_callMethod = (...args) => super.method(...args);
    return _gen.apply(this, arguments);
  }
}

let n = new A(10);
let m = new A(100);
n.gen(m).next(1); // 107 101
```